### DISCUSSION TITLE
is it possible to attach elements on top of the link

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#2149] [Q&A]
### Introduction

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/117301232/234305899-7a62944f-de6d-417f-b4c8-6ebb08123185.gif)

is it possible to attach elements on top of the link ( if it is near to the link ) when I drop elements to the canvas

![Screenshot 2023-04-25 194241](https://user-images.githubusercontent.com/117301232/234305345-1d9e9542-14d1-4929-bd9e-fabd7721ca49.png)

like this

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Elements can not be attached to links. But labels can (i.e. you can add a label to the link and remove the element on drop).

Here's an example of dropping elements on links: https://jsfiddle.net/kumilingus/ypc9uwjs/
In the example, we look at the `evt.target` under the pointer. If you want to highlight the link when you are in the link's proximity, you can look for an intersection or the distance from the closest point on the link (use [linkView.getConnection()](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#dia.LinkView.prototype.getConnection) to get the geometric representation of the link).

It also depends on how you use links, but maybe you can use elements that look like a line and use the built-in embedding feature.

**Reply by reshin-pulapaddy** [reply]
Hi @kumilingus 

In touch devices, the above fiddle does not work. I need this functionality in the touch devices as well.

**Reply by kumilingus** [reply]
You can not use `evt.target` on touch devices (it always refers to the element from `touchstart`).
```js
const view = paper.findView(evt.target);
```

You need to use `document.elementFromPoint()` instead.
```js
const view = paper.findView(document.elementFromPoint(evt.clientX, evt.clientY));
```

Update Fiddle [here](https://jsfiddle.net/kumilingus/ypc9uwjs/).

**Reply by reshin-pulapaddy** [reply]
Thank you

**Reply by reshin-pulapaddy** [reply]
@kumilingus 
is there any option to add the same functionality when I drag an element from stencil paper to paper

**Reply by kumilingus** [reply]
Yes. There are various Stencil events that you can listen to (`element:dragstart`, `element:drag`, `element:dragend`, `element:drop`).
Here are examples:
https://www.jointjs.com/demos/drop-stencil-element-as-shape-icon
https://www.jointjs.com/demos/planogram
